src/segmented-button/segmented-button/foundation.ts

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useRef } from 'react';
+import { useFoundation } from '@rmwc/base';
+import {
+  MDCSegmentedButtonFoundation,
+  SegmentDetail
+} from '@material/segmented-button';
+import { SegmentedButtonProps } from './';
+import { SegmentApi } from '../segment';
+
+export const useSegmentedButtonFoundation = (
+  props: SegmentedButtonProps & React.HTMLProps<any>
+) => {
+  const segmentApi = useRef<SegmentApi>();
+  const setSegmentApi = (api: SegmentApi) => (segmentApi.current = api);
+  const segmentsList = useRef<SegmentApi[]>([]);
+
+  const mappedSegments = segmentsList.current.map(
+    (segment: SegmentApi, index: number) => {
+      return {
+        index,
+        selected: segment.isSelected(),
+        segmentId: segment.getSegmentId()
+      };
+    }
+  );
+
+  const { foundation, ...elements } = useFoundation({
+    props,
+    elements: { rootEl: true },
+    foundation: ({ rootEl, emit }) => {
+      return new MDCSegmentedButtonFoundation({
+        hasClass: (className) => {
+          return rootEl.hasClass(className);
+        },
+        getSegments: () => {
+          console.log({ mappedSegments });
+          return mappedSegments;
+        },
+        selectSegment: (indexOrSegmentId) => {
+          const segmentDetail = mappedSegments.find(
+            (detail: SegmentDetail) =>
+              detail.index === indexOrSegmentId ||
+              detail.segmentId === indexOrSegmentId
+          );
+          console.log({ segmentDetail });
+          if (segmentDetail) {
+            segmentsList.current[segmentDetail.index].setSelected();
+          }
+        },
+        unselectSegment: (indexOrSegmentId) => {
+          const segmentDetail = mappedSegments.find(
+            (detail: SegmentDetail) =>
```

Comment:
Looks like this duplicates the code at line 41, this should be refactored to remove the code duplication